src/System.Net.Security/tests/FunctionalTests/KerberosTest.cs

```diff
@@ -0,0 +1,331 @@
+                Assert.Equal(TokenImpersonationLevel.Identification, client.ImpersonationLevel);
+                Assert.True(client.IsEncrypted);
+                Assert.True(client.IsMutuallyAuthenticated);
+                Assert.False(client.IsServer);
+                Assert.True(client.IsSigned);
+                Assert.False(client.LeaveInnerStreamOpen);
+
+                IIdentity serverIdentity = client.RemoteIdentity;
+                Assert.Equal("Kerberos", serverIdentity.AuthenticationType);
+                Assert.True(serverIdentity.IsAuthenticated);
+                IdentityValidator.AssertHasName(serverIdentity, target);
+            }
+        }
+
+        [Fact, OuterLoop]
+        [PlatformSpecific(PlatformID.Linux | PlatformID.OSX)]
+        public void NegotiateStream_StreamToStream_KerberosAuthInvalidUser_Failure()
+        {
+            if (!_isKrbAvailable)
+            {
+                return;
+            }
+
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var client = new UnixGssFakeNegotiateStream(clientStream))
+            {
+                Assert.False(client.IsAuthenticated);
+
+                string user = string.Format("{0}@{1}", TestConfiguration.KerberosUser, TestConfiguration.Realm);
+                string target = string.Format("{0}@{1}", TestConfiguration.HostTarget, TestConfiguration.Realm);
+                NetworkCredential credential = new NetworkCredential(user.Substring(1), TestConfiguration.Password);
+                Assert.Throws<AuthenticationException>(() =>
+                {
+                    client.AuthenticateAsClientAsync(credential, target);
+                });
+            }
+        }
+
+        [Fact, OuterLoop]
+        [PlatformSpecific(PlatformID.Linux | PlatformID.OSX)]
+        public void NegotiateStream_StreamToStream_KerberosAuthInvalidPassword_Failure()
+        {
+            if (!_isKrbAvailable)
+            {
+                return;
+            }
+
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var client = new UnixGssFakeNegotiateStream(clientStream))
+            {
+                Assert.False(client.IsAuthenticated);
+
+                string user = string.Format("{0}@{1}", TestConfiguration.KerberosUser, TestConfiguration.Realm);
+                string target = string.Format("{0}@{1}", TestConfiguration.HostTarget, TestConfiguration.Realm);
+                NetworkCredential credential = new NetworkCredential(user, TestConfiguration.Password.Substring(1));
+                Assert.Throws<AuthenticationException>(() =>
+                {
+                    client.AuthenticateAsClientAsync(credential, target);
+                });
+            }
+        }
+
+        [Fact, OuterLoop]
```

Comment:
> Are these [OuterLoop] attributes just a temporary thing so that PR testing doesn't break

This was also an offline recommendation from @Priya91 and @joshfree since there are prerequisites like test user being able to sudo without password.